### ISSUE TITLE
no-curly-component-invocation creates false positives with default `octane` preset

### ISSUE THREAD

**Issue by Turbo87** [ember-template-lint/ember-template-lint#1026]
The `octane` preset is using `'no-curly-component-invocation': { requireDash: false }`, which appears to cause a lot of false positives:

```
/Users/tbieniek/Code/crates.io/app/templates/me/following.hbs
  12:24  error  You are using the component {{currentPageStart}} with curly component syntax. You should use <CurrentPageStart> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['currentPageStart'] }`.  no-curly-component-invocation
  12:47  error  You are using the component {{currentPageEnd}} with curly component syntax. You should use <CurrentPageEnd> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['currentPageEnd'] }`.  no-curly-component-invocation
  13:29  error  You are using the component {{totalItems}} with curly component syntax. You should use <TotalItems> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['totalItems'] }`.  no-curly-component-invocation
  22:8  error  You are using the component {{currentSortBy}} with curly component syntax. You should use <CurrentSortBy> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['currentSortBy'] }`.  no-curly-component-invocation

/Users/tbieniek/Code/crates.io/app/templates/me/index.hbs
  16:10  error  You are using the component {{model.user.name}} with curly component syntax. You should use <Model.User.Name> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['model.user.name'] }`.  no-curly-component-invocation
  18:10  error  You are using the component {{model.user.login}} with curly component syntax. You should use <Model.User.Login> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. `'no-curly-component-invocation': { allow: ['model.user.login'] }`.  no-curly-component-invocation
```

**Comment by rwjblue** [comment]
@Turbo87 - All of those examples should have _also_ printed an error for [no-implicit-this](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-implicit-this.md).

**Comment by rwjblue** [comment]
Also, if those paths were not flagged by `no-implicit-this`, can you file a bug about that specifically?

**Comment by Turbo87** [comment]
> Also, if those paths were not flagged by `no-implicit-this`, can you file a bug about that specifically?

they very likely were flagged, but I disabled that rule, because I hadn't run the corresponding codemod on the code yet.

**Comment by rwjblue** [comment]
Gotcha, this bug report makes much more sense now. 😸
